woocommerce/wc-admin
# Refactor menu item mapping and sorting

Fixes #6346 

Refactors the menu utilities to be more performant and include sorting.

### Detailed test instructions:

1. Activate multiple extensions that register WC menu items.
2. Favorite and unfavorite extensions
3. Check that the menu still works as expected.
4. Check that the order of menu items is correct in the extensions menu after unfavoriting.
5. Check that a user without permission to access a menu item cannot see said menu item.
